lib/fibonacci.rb

```diff
@@ -1,8 +1,24 @@
 # Improved Fibonacci
+# Dynamic Programming Recursive Solution
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 # Hint, you may want a recursive helper method
+
 def fibonacci(n)
```

Comment:
This works, but you're maintaining the entire list of fibonacci numbers from 0 to n.  You only need to keep the last two fibonacci numbers.